### PR TITLE
Add put and delete functionality

### PR DESCRIPTION
**Pull request by buzz1710**
- Use cURL for http requests
- Add PUT and DELETE functionality
- UTF-8 encode post/put data
- Use tabs for UI

screen shot 2016-03-22 at 14 07 53

### CODE REVIEW

File: callback.php [hunk 1]
```diff
@@ -7,10 +7,9 @@
 
 /* Exchange the authorisation code for an access_token */
 $response = $sageone_client->getAccessToken($_GET['code']);
-$access_token = json_decode($response, true)['access_token'];
 
 /* redirect with the access_token */
```
Comment (chrisbarber86) [inline callback.php]: Comment should be updated

File: callback.php [hunk 2]
```diff
@@ -7,10 +7,9 @@
 
 /* Exchange the authorisation code for an access_token */
 $response = $sageone_client->getAccessToken($_GET['code']);
-$access_token = json_decode($response, true)['access_token'];
 
-/* redirect with the access_token */
-header("Location: http://localhost:8000/sageone_data.php?access_token=" . $access_token);
+/* redirect with the response */
+header("Location: http://localhost:8000/sageone_data.php?token_response=" . $response);
```
Comment (tosch) [inline callback.php]: You really should urlencode `$response`, or is `$sageone_client->getAccessToken($_GET['code'])` doing that?

File: sageone_data.php [hunk 3]
```diff
@@ -22,27 +28,60 @@
...
+          <li role="presentation"><a href="#put" aria-controls="put" role="tab" data-toggle="tab">PUT</a></li>
+          <li role="presentation"><a href="#delete" aria-controls="delete" role="tab" data-toggle="tab">DELETE</a></li>
+        </ul>
+        <div class="tab-content">
+          <div role="tabpanel" class="tab-pane fade in active" id="get">
+              <h3>GET request</h3>
+              <form action="sageone_response.php" method="get" class="endpoint_form">
+                <label for="get_endpoint">Endpoint</label>
+                <input id="get_endpoint" name="get_endpoint" type="text" value="" class="form-control" required="true">
+                <p>Example: accounts/v1/contacts</p>
```
Comment (tosch) [inline sageone_data.php:44]: `v3`?

File: sageone_data.php [hunk 4]
```diff
@@ -22,27 +28,60 @@
...
+                <input id="get_access_token" name="get_access_token" type="hidden" value="<?php echo $access_token ?>">
+                <input type='submit' value='GET' class='btn btn-primary'>
+              </form>
+          </div>
+          <div role="tabpanel" class="tab-pane fade" id="post">
+            <h3>POST request</h3>
+            <form action="sageone_response.php" method="post" class="endpoint_form">
+              <label for="post_endpoint">Endpoint</label>
+              <input id="post_endpoint" name="post_endpoint" type="text" value="" class="form-control" required="true">
+              <p>Example: accounts/v1/contacts</p>
```
Comment (tosch) [inline sageone_data.php]: `v3`?

File: sageone_data.php (hunk 3 above)
Comment (buzz1710) [inline sageone_data.php:44]: this PR targets master which is for v1/v2. I've updated this in the v3 branch

**Comment by tosch** [comment]
I'm not sure if we should use the PHP cUrl extension as it might not be available to everyone (maybe we could use the HTTP client in the Zend Framework?), but anyway, I'm far from being a PHP expert. So, for me it's a 👍  (after encoding the response in the redirection)

File: sageone_data.php [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
+/* prettify JSON response for readability */
+$json = json_decode($response);
+$pretty_json = json_encode($json, JSON_PRETTY_PRINT);
 ?>
 <html>
   <head>
     <title>Authorize with Sage One</title>
     <link type="text/css" rel="stylesheet" href="sample_app.css">
+    <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
```
Comment (fejao) [inline sageone_data.php]: ? http://code.jquery.com/jquery-latest.min.js ?
or do we need a fixed version?
😎

**Comment by fejao** [comment]
👍

**Pull request merged**